[DOC-124] Categories for YB-Voyager CLI (#19325)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-voyager/reference/data-migration/import-data.md

    parent: data-migration
    weight: 60
type: docs
---

This page describes the usage of the following import commands:

- [import data](#import-data)
- [import data status](#import-data-status)

## import data

For offline migration, [Import the data](../../../migrate/migrate-steps/#import-data) to the YugabyteDB database.

For live migration (and fall-forward), the command [imports the data](../../../migrate/migrate-steps/#import-data) from the `data` directory to the target database, and starts ingesting the new changes captured by `export data` to the target database.

### Syntax

```text
yb-voyager import data [ <arguments> ... ]
```

#### Arguments

The valid *arguments* for import data are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |
| --batch-size <number> | Size of batches in the number of rows generated for ingestion during import data. (default: 20000 rows)<br>Example: `yb-voyager import data ... --batch-size 20000` |
| --continue-on-error | Ignores the error while executing the DDLs for resuming sequences on target db after the data is imported. (default: false) |

> | --disable-pb | Use this argument to not display progress bars. For live migration, `--disable-pb` can also be used to hide metrics for import data. (default: false) |

Reviewer: ```suggestion
| --disable-pb | Turn off the display of progress bars. For live migration, `--disable-pb` can also be used to hide metrics for import data. (default: false) |
```

---
title: import data reference
headcontent: yb-voyager import data
linkTitle: import data
description: YugabyteDB Voyager import data reference
menu:
  preview_yugabyte-voyager:
    identifier: voyager-import-data
    parent: data-migration
    weight: 60
type: docs
---

This page describes the usage of the following import commands:

- [import data](#import-data)
- [import data status](#import-data-status)

## import data

For offline migration, [Import the data](../../../migrate/migrate-steps/#import-data) to the YugabyteDB database.

For live migration (and fall-forward), the command [imports the data](../../../migrate/migrate-steps/#import-data) from the `data` directory to the target database, and starts ingesting the new changes captured by `export data` to the target database.

### Syntax

```text
yb-voyager import data [ <arguments> ... ]
```

#### Arguments

The valid *arguments* for import data are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |

> | --batch-size <number> | Size of batches generated for ingestion during import data. (default: 20000 rows) |

Author: @priyanshi-yb - Is <number> number of rows ? where default is 20000 rows. 
I am unclear too. Can we add an example?

> | --batch-size <number> | Size of batches generated for ingestion during import data. (default: 20000 rows) |

Reviewer 2: yes this parameter accepts an integer which is number of rows. 
example - 
`yb-voyager import data ... --batch-size 20000`

        --target-db-schema schemaName \ # MySQL and Oracle only
        --parallel-jobs connectionCount
```

## import data status

For offline migration, get the status report of an ongoing or completed data import operation. The report contains migration status of tables, number of rows or bytes imported, and percentage completion.

For live migration, get the status report of import data. For live migration with fall forward, the report also includes the status of fall forward setup. The report includes the status of tables, the number of rows imported, the total number of changes imported, the number of `INSERT`, `UPDATE`, and `DELETE` events, and the final row count of the target or fall-forward database.

### Syntax

```text
Usage: yb-voyager import data status [ <arguments> ... ]
```

#### Arguments

The valid *arguments* for import data status are described in the following table:

| Argument | Description/valid options |
| :------- | :------------------------ |
| -e, --export-dir <path> | Path to the export directory. This directory is a workspace used to keep the exported schema, data, state, and logs.|
| -h, --help | Command line help. |

> | target-db-password | Password of the target database. Live migrations only. |

Reviewer 2: change description - 
```
Target database password. Alternatively, you can also specify the password by setting the environment variable `TARGET_DB_PASSWORD`. If you don't provide a password via the CLI during any migration phase, yb-voyager will prompt you at runtime for a password. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose the password in single quotes. Required for live migrations only.
```

> | ff-db-password | Password of the fall-forward database. Live migration with fall-forward only. |

Reviewer 2: ```
Password to connect to the fall-forward database server. Alternatively, you can also specify the password by setting the environment variable `FF_DB_PASSWORD`. If you don't provide a password via the CLI or environment variable during any migration phase, yb-voyager will prompt you at runtime for a password. If the password contains special characters that are interpreted by the shell (for example, # and $), enclose the password in single quotes. Required for live migration with fall forward only.
```
